haas/model.py

```diff
@@ -40,17 +42,30 @@
 
 db = SQLAlchemy(app)
 
+# Sets up variant type so that postgresql can use BIGINT primary keys
+# while sqlite uses Integer primary keys.
+BigIntegerType = BigInteger().with_variant(
+        sqlite.INTEGER(), 'sqlite')
+
 
 def init_db(uri=None):
     """Start up the DB connection.
 
-    `uri` is the uri to use for the databse. If it is None, the uri from the
+    `uri` is the uri to use for the database. If it is None, the uri from the
     config file will be used.
     """
     if uri is None:
         uri = cfg.get('database', 'uri')
     app.config.update(SQLALCHEMY_DATABASE_URI=uri)
 
+    global BigIntegerType
+
+    # Decide if primary keys should be BIGINTs or Integers
+    if 'postgresql' in uri:
```

Comment:
That seems strange.

Maybe try debugging it to see what's happening?

Perhaps there could be a bug in our version of SQLAlchemy? You could try bumping the package version for flask-sqlalchemy to <3.0, since there's a 2.2 out now (though be sure to look for problems, like extra messages in the unit tests).